test_programs/plonky2_trace/3_two_files/expected_trace.json

```diff
@@ -53,6 +48,10 @@
   { "Value": { "variable_id": 0, "value": { "kind": "Int", "i": 1, "type_id": 1 } } },
   { "Value": { "variable_id": 1, "value": { "kind": "Int", "i": 2, "type_id": 1 } } },
   { "Return": { "return_value": { "kind": "None", "type_id": 2 } } },
+  { "Step": { "path_id": 1, "line": 7 } },
+  { "Value": { "variable_id": 0, "value": { "kind": "Int", "i": 1, "type_id": 1 } } },
+  { "Value": { "variable_id": 1, "value": { "kind": "Int", "i": 2, "type_id": 1 } } },
+  { "Value": { "variable_id": 2, "value": { "kind": "Int", "i": 3, "type_id": 1 } } },
```

Comment:
Why are those values added? I am not able to conclude if this is an expected behavior based on the provided description.

Reply:
This comes from line 124 in the code; note that the logic is "add a step after Return; add variables (for that step)". These are the variables for the step.

Reply:
Github has some issues showing the code differences of the commits and showed me this piece of code as a part of the "Fix steps before call" commit. That's where my confusion came from.

Reply:
I see, so it's actually unrelated to the return code.

It's an accidental bug-fix :) Before, we could (and indeed, we did) miss a step if the line performing the function call had additional operations. In this case, the line is `let w = foo(x, y)`. The code was just comparing if the new stack of locations has a suffix that's different than the previous stack. Only if so, the steps from this suffix were appended to the trace.

The new code always adds exactly one step (except in the case where there are no stack_frames). This is the case, because the heuristic for whether the code reaches a new source location is (and has been) implemented in `step_debugger()`. So there's no need to implement another heuristic here. Keeping the new simple logic is not only enough, but also better.

Reply:
Good, feel free to rebase and merge.